app/utils/sagaInjectors.js

```diff
@@ -25,39 +25,46 @@ const checkDescriptor = descriptor => {
 
 export function injectSagaFactory(store, isValid) {
   return function injectSaga(key, descriptor = {}, args) {
-    if (!isValid) checkStore(store);
+    return new Promise((resolve, reject) => {
+      try {
+        if (!isValid) checkStore(store);
 
-    const newDescriptor = {
-      ...descriptor,
-      mode: descriptor.mode || DAEMON,
-    };
-    const { saga, mode } = newDescriptor;
+        const newDescriptor = {
+          ...descriptor,
+          mode: descriptor.mode || RESTART_ON_REMOUNT,
+        };
+        const { saga, mode } = newDescriptor;
 
-    checkKey(key);
-    checkDescriptor(newDescriptor);
+        checkKey(key);
+        checkDescriptor(newDescriptor);
 
-    let hasSaga = Reflect.has(store.injectedSagas, key);
+        let hasSaga = Reflect.has(store.injectedSagas, key);
 
-    if (process.env.NODE_ENV !== 'production') {
-      const oldDescriptor = store.injectedSagas[key];
-      // enable hot reloading of daemon and once-till-unmount sagas
-      if (hasSaga && oldDescriptor.saga !== saga) {
-        oldDescriptor.task.cancel();
-        hasSaga = false;
-      }
-    }
+        if (process.env.NODE_ENV !== 'production') {
+          const oldDescriptor = store.injectedSagas[key];
+          // enable hot reloading of daemon and once-till-unmount sagas
+          if (hasSaga && oldDescriptor.saga !== saga) {
+            oldDescriptor.task.cancel();
+            hasSaga = false;
+          }
+        }
 
-    if (
-      !hasSaga ||
-      (hasSaga && mode !== DAEMON && mode !== ONCE_TILL_UNMOUNT)
-    ) {
-      /* eslint-disable no-param-reassign */
-      store.injectedSagas[key] = {
-        ...newDescriptor,
-        task: store.runSaga(saga, args),
-      };
-      /* eslint-enable no-param-reassign */
-    }
+        if (
+          !hasSaga ||
+          (hasSaga && mode !== DAEMON && mode !== ONCE_TILL_UNMOUNT)
+        ) {
+          /* eslint-disable no-param-reassign */
+          store.injectedSagas[key] = {
+            ...newDescriptor,
+            task: store.runSaga(saga, args),
+          };
+          /* eslint-enable no-param-reassign */
+        }
+        resolve();
+      } catch (error) {
+        reject(error);
```

Comment:
You have a `reject` here, which makes sense. But no error handling in the `injectSaga` HOC. I think that should be added.